parquet-tools/src/test/java/org/apache/parquet/tools/command/TestPruneColumnsCommand.java

```diff
@@ -0,0 +1,258 @@
+import static org.apache.parquet.schema.Type.Repetition.REPEATED;
+import static org.apache.parquet.schema.Type.Repetition.REQUIRED;
+import static org.junit.Assert.assertEquals;
+
+public class TestPruneColumnsCommand {
+
+  private static final String root = "target/";
+  private final int numRecord = 1000;
+  private PruneColumnsCommand command = new PruneColumnsCommand();
+  private Configuration conf = new Configuration();
+  private Random rand = new Random();
+
+  @Test
+  public void testPruneOneColumn() throws Exception {
+    // Create Parquet file
+    String inputFile = createParquetFile();
+    String outputFile = getParquetFileName();
+
+    // Remove column
+    String cargs[] = {inputFile, outputFile, "Gender"};
+    executeCommandLine(cargs);
+
+    // Verify the schema are not changed for the columns not pruned
+    ParquetMetadata pmd = ParquetFileReader.readFooter(conf, new Path(outputFile), ParquetMetadataConverter.NO_FILTER);
+    MessageType schema = pmd.getFileMetaData().getSchema();
+    List<Type> fields = schema.getFields();
+    assertEquals(fields.size(), 3);
+    assertEquals(fields.get(0).getName(), "DocId");
+    assertEquals(fields.get(1).getName(), "Name");
+    assertEquals(fields.get(2).getName(), "Links");
+    List<Type> subFields = fields.get(2).asGroupType().getFields();
+    assertEquals(subFields.size(), 2);
+    assertEquals(subFields.get(0).getName(), "Backward");
+    assertEquals(subFields.get(1).getName(), "Forward");
+
+    // Verify the data are not changed for the columns not pruned
+    List<String> prunePaths = Arrays.asList("Gender");
+    validateColumns(inputFile, prunePaths);
+  }
+
+  @Test
+  public void testPruneMultiColumns() throws Exception {
+    // Create Parquet file
+    String inputFile = createParquetFile();
+    String outputFile = getParquetFileName();
+
+    // Remove columns
+    String cargs[] = {inputFile, outputFile, "Name", "Gender"};
+    executeCommandLine(cargs);
+
+    // Verify the schema are not changed for the columns not pruned
+    ParquetMetadata pmd = ParquetFileReader.readFooter(conf, new Path(outputFile), ParquetMetadataConverter.NO_FILTER);
+    MessageType schema = pmd.getFileMetaData().getSchema();
+    List<Type> fields = schema.getFields();
+    assertEquals(fields.size(), 2);
+    assertEquals(fields.get(0).getName(), "DocId");
+    assertEquals(fields.get(1).getName(), "Links");
+    List<Type> subFields = fields.get(1).asGroupType().getFields();
+    assertEquals(subFields.size(), 2);
+    assertEquals(subFields.get(0).getName(), "Backward");
+    assertEquals(subFields.get(1).getName(), "Forward");
+
+    // Verify the data are not changed for the columns not pruned
+    List<String> prunePaths = Arrays.asList("Name", "Gender");
+    validateColumns(inputFile, prunePaths);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testNotExistsColumn() throws Exception {
```

Comment:
Let's check the following scenario.
User wants to prune the columns `a`, `b` and `c` from several parquet files. The schema of the files are not the same because they evolved (e.g. column `c` is added after some files were already created or column `a` is used for partitioning so many files do not contain it).
If you think the previous scenario is valid then the tool should not throw an exception in case a column to be pruned is not in the file. Meanwhile, the user might mistyped the column name and it may not notify. So, it is your choice how you should handle this case. :)

Reply:
That is a really good point! I was trying to prevent 'silent failing' when user has a typo for the column names. Now I think we can just print a warning message.

Reply:
You still expects the exception to be thrown in the two related tests. That's why Travis fails.

Reply:
Thanks for pointing out! Just fixed both.